src/inventory/player_inventory.rs

```diff
@@ -213,39 +199,66 @@ impl PlayerInventory {
                 size,
             );
         }
+        let x_offset = -(size * 4.5);
+        let screen_height = renderer.screen_data.read().safe_height as f64;
+        let y_offset = (screen_height / scale) / 2.0;
+        self.base_inventory.clone().write().update_offset(
+            x_offset / size,
+            (y_offset + 59.0) / 16.0,
+            renderer,
+        );
         self.dirty = true;
     }
-
-    pub(crate) fn get_raw_slot_mut(&mut self, idx: u16) -> &mut Slot {
-        self.slots.get_mut(idx as usize).unwrap()
-    }
 }
 
 impl Inventory for PlayerInventory {
     fn size(&self) -> u16 {
-        self.slots.len() as u16
+        let mut size = self.slots.len() as u16;
+        size += self.base_inventory.read().size();
+        if self.offhand_slot.is_some() {
+            size += 1;
+        }
+        size
     }
 
     fn id(&self) -> i32 {
-        -1
+        0
```

Comment:
are you sure that passing 0 here is correct, as far as i remember there was a good reason i used -1 instead of 0, i can't remember why anymore tho

Reply:
When I tested this using -1, inventory operations didn't work. It only started working when I switched to 0. The wiki also states the player inventory is 0: https://wiki.vg/Protocol#Set_Container_Content

Reply:
Fair enough, could you add a comment with that link? (it would be cool if you could add comments to every parts of the code you edited that have smth todo with protocol specifics)

Reply:
I've added a comment with a link to the docs.